### ISSUE TITLE
meshservice: Enable linking of Outbounds in DataPlaneViz to MeshService detail page

### ISSUE THREAD

**Issue by johncowen** [kumahq/kuma-gui#2848]
### Description

For the dataplane overview viz we use Envoy stats as the data source to generate the viz/GUI, current this generates the following Outbounds:

![Screenshot 2024-09-02 at 13 56 39](https://github.com/user-attachments/assets/cba4d606-ec80-42dd-9243-61f81e40c37f)

The links here take us to the following URL:

http://localhost:5681/gui/meshes/default/data-planes?s=service:demo-app_kuma-demo_msvc_5000

i.e. Show me the list of dataplanes that have the `kuma.io/service` equalling `demo-app_kuma-demo_msvc_5000`, which of course don't exist. Therefore we show an empty list.

The cards in the above screenshot should instead read (ideally) `demo-app`, and link to:

http://localhost:5681/gui/meshes/default/services/mesh-services/demo-app.kuma-demo/overview

Note: In some environments the name used in the above screenshot can contain the hash used for identifying dataplanes in the backend i.e. `redis-2a3s4d5f_kuma-demo_msvc_5000`.

---

Overall this presents a problem because it "looks like" we can only use `-` and `_` as separators/markers, which are also valid user values, meaning any sort of parsing/regexing we do is not going to be guaranteed to work correctly.

**Comment by jakubdyszkiewicz** [comment]
> Triage: it requires backend work, because of name hashes. For upcoming release let's just not display the link.

**Comment by johncowen** [comment]
PR to not display the link for now:

https://github.com/kumahq/kuma-gui/pull/2853

**Comment by johncowen** [comment]
FYI #2853 is in, I'm guessing we just need to bump the milestone here seeing as it sounds like we need backend support which won't come until after the upcoming release (see triage comment)

I'll leave 2.9 for now but guessing we can bump that when the milestone gets reviewed.

**Comment by lahabana** [comment]
@johncowen to search for a backend issue to relate to this.

**Comment by johncowen** [comment]
@jakubdyszkiewicz do you have any insight on the backend work related to this issue? Not sure if there is already an issue, or whether you had any thoughts on whats required for the backend support we need for this?

**Comment by lahabana** [comment]
@johncowen is this what you are looking for: https://github.com/kumahq/kuma/pull/11865

**Comment by johncowen** [comment]
I don't think so but not totally sure, shall we remove the accepted label so we can discuss in triage? I won't be working this until way after the next triage anyway. If not, we could also just chat through this next time we sync.

**Comment by lobkovilya** [comment]
> Triage: waiting on https://github.com/kumahq/kuma/issues/12044

**Comment by lukidzi** [comment]
xref: https://github.com/kumahq/kuma/issues/13256
